### PR TITLE
fix ownProperty

### PR DESCRIPTION
**Pull request by yiminghe**
SymNew.hasOwnProperty('for') should be true

### CODE REVIEW

**Review by ljharb — CHANGES_REQUESTED** [review]
Can you provide a regression test?

Separately, the setProto is absolutely critical so the inheritance chain of Symbols is correct, so it must not be deleted.

**Comment by yiminghe** [comment]
> Can you provide a regression test?
> 
> Separately, the setProto is absolutely critical so the inheritance chain of Symbols is correct, so it must not be deleted.

I didn't delete setProto, it's inside copyStaticProperties

**Comment by ljharb** [comment]
ah, thanks, just missed it.

File: test/shimmed.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 'use strict';
 
+// for test:
+
+// modify es-abstract/helpers/getInferredName = null
+
+// delete global.Symbol.prototype.description;
+
```
Comment (ljharb) [inline test/shimmed.js]: i don't think we need these comments here; CI will run in appropriate node versions to exercise it.
```suggestion
```

File: shim.js [hunk 2]
```diff
@@ -16,6 +16,11 @@ var define = function defineGetter(getter) {
 	});
 };
 
+var copyStaticProperties = function copyStatProperties(ctorNew, ctorOld) {
+	setProto(ctorNew, Object.getPrototypeOf(ctorOld));
+	Object.defineProperties(ctorNew, Object.getOwnPropertyDescriptors(ctorOld));
```
Comment (ljharb) [inline shim.js]: - `Object.getPrototypeOf` may not exist in every engine, and either way, it needs to be cached at module level
 - `Object.defineProperties` needs to be cached at module level
 - `Object.getOwnPropertyDescriptors` should instead use https://www.npmjs.com/package/object.getownpropertydescriptors

File: test/shimmed.js [hunk 3]
```diff
@@ -32,6 +38,11 @@ test('shimmed', function (t) {
 		st.end();
 	});
 
+	t.test('hasOwnProperty', function (st) {
+		st.ok(Object.prototype.hasOwnProperty.call(Symbol, 'for'), 'for is Symbol\' own property');
```
Comment (ljharb) [inline test/shimmed.js]: Are there other properties that should be checked in this way? Perhaps `keyFor`, and all the well-known symbols as well?

**Review by ljharb — CHANGES_REQUESTED** [review]

File: shim.js (hunk 2 above)
Comment (yiminghe) [reply shim.js]: cache is not needed, shim will run only once

Comment (ljharb) [reply shim.js]: Caching is always needed; and no, the shim method can run many times. The file will only be *required* once.

File: test/shimmed.js [hunk 4]
```diff
@@ -32,6 +32,18 @@ test('shimmed', function (t) {
...
 
+	t.test('hasOwnProperty', function (st) {
+		const ownProperties = ['for', 'keyFor', 'asyncIterator', 'hasInstance', 'isConcatSpreadable', 'iterator', 'match', 'matchAll', 'replace', 'search', 'species', 'split', 'toPrimitive', 'toStringTag', 'unscopables', 'observable'];
+		// some old engine may miss
+		let atLeastOne = false;
+		for (var i = 0; i < ownProperties.length; i++) {
+			var p = ownProperties[i];
+			atLeastOne = atLeastOne || Object.prototype.hasOwnProperty.call(Symbol, p);
+		}
+		st.ok(atLeastOne, 'preserve Symbol\' own property');
```
Comment (ljharb) [inline test/shimmed.js:43]:
```suggestion
		for (var i = 0; i < ownProperties.length; i++) {
			var p = ownProperties[i];
			st.ok(Object.prototype.hasOwnProperty.call(Symbol, p), 'has own property: ' + p);
		}
```

File: test/shimmed.js [hunk 5]
```diff
@@ -32,6 +32,18 @@ test('shimmed', function (t) {
 		st.end();
 	});
 
+	t.test('hasOwnProperty', function (st) {
+		const ownProperties = ['for', 'keyFor', 'asyncIterator', 'hasInstance', 'isConcatSpreadable', 'iterator', 'match', 'matchAll', 'replace', 'search', 'species', 'split', 'toPrimitive', 'toStringTag', 'unscopables', 'observable'];
```
Comment (ljharb) [inline test/shimmed.js:36]:
```suggestion
		var ownProperties = ['for', 'keyFor', 'asyncIterator', 'hasInstance', 'isConcatSpreadable', 'iterator', 'match', 'matchAll', 'replace', 'search', 'species', 'split', 'toPrimitive', 'toStringTag', 'unscopables', 'observable'];
```

**Review by ljharb — CHANGES_REQUESTED** [review]

File: test/shimmed.js (hunk 5 above)
Comment (ljharb) [reply test/shimmed.js:36]: also, `observable` is not a thing, unless you've perhaps installed a polyfill for the long-defunct stage 2 Observables proposal

**Comment by ljharb** [comment]
Unfortunately, you deleted your fork, so this can't ever be rebased again :-(

I'll adapt your test with a better fix and land it shortly.

**Pull request closed without merging**